system-addon/content-src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -24,9 +24,23 @@ class ContextMenu extends React.Component {
   componentDidUnmount() {
     window.removeEventListener("click", this.hideContext);
   }
-  onKeyDown(index) {
-    if (index > this.props.tabbableOptionsLength) {
-      this.hideContext();
+  onKeyDown(event, option) {
+    switch (event.key) {
+      case "Tab":
+        if (event.shiftKey) {
+          if (option.first) {
+            this.hideContext();
+          }
+          return;
+        }
+        if (option.last) {
+          this.hideContext();
+        }
+        break;
+      case "Enter":
+        this.hideContext();
+        option.onClick();
```

Comment:
Yeah I'm not sure about this, but I think it's ok - the ```onClick``` has the same ordering and so far it hasn't been a problem. I think since it's calling into a function that's defined by it's props, it should still execute the function regardless of visibility